src/card_searcher.rs

```diff
@@ -70,13 +71,20 @@ impl Search for &str {
 /// in a scryfall search parameters. The valid parameters can be seen
 /// [here](https://scryfall.com/docs/syntax).
 pub trait Param: fmt::Debug {
+    /// Adds a parameter's string version to the passed string
```

Comment:
optional: I support this change overall, but I'm noticing it looks remarkably similar to how `fmt::Display` works. What would you think of defining `Param` as follows:
```rust
pub trait Param: fmt::Display + fmt::Debug {}
```
Then the logic from `to_param` could be implemented in `fmt::Display::fmt`.

If you prefer, I'd be in favor of merging this PR and making that change in a separate one.

Reply:
I did do that because I somehow thought doing that made all `Display` also `Param` but that's dumb. I'm gonna make that change.

Reply:
Hmm.. I ran into a problem, the `BorderColor` enum already has a display implementation so I can't make the custom one here to make it a param. :thinking:

Reply:
Hm, I see what you mean. How about we submit it as you have. Maybe it'd be better to leave this problem for the next version of this API.

This is sort of what I have in mind.

```rust
pub enum SearchExpr {
    And(Vec<SearchExpr>),
    Or(Vec<SearchExpr>),
    Not(Box<SearchExpr>),
    Param(SearchParam),
}

pub enum SearchParam {
    Property(PropertyKind), // equivalent of BoolParam right now
    Value(ValueKind, Box<dyn ParamValue>), // ParamValue would be implemented for String, Vec<Color>, BorderColor, etc
    Compare(ComparisonOp, ValueKind, Box<dyn ParamValue>), // Perhaps this could be constrained a bit more.
}
```

One downside here is that the enum variant constructors would need to be replaced with static methods on SearchParam or SearchExpr.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=96426bc9a819c4b103a183d6b687ca75

The overloading is an interesting possibility... or a bad idea.

Reply:
oh boy, that overloading sure is spicy, maybe use the `&&` and the `||` operators instead? Since they are more closely linked with logical `and` and `or`?

I think it's cool enough to give it a go tbh. And it's optional too, as long as we keep the normal methods too. Also maybe use `VecDeque` instead of `Vec` just to optimize the insertions at the head?

Reply:
You can't overload the lazy boolean operators, unfortunately. That would have been my first choice. Nice idea with VecDeque.